dcos_launch/config.py

```diff
@@ -127,6 +127,7 @@ def get_validated_config(user_config: dict, config_dir: str) -> dict:
     # validate against the fields common to all configs
     user_config['config_dir'] = config_dir
     validator = LaunchValidator(COMMON_SCHEMA, config_dir=config_dir, allow_unknown=True)
+    user_config.setdefault('dcos_config', {})
```

Comment:
Why move this? I believe dcos_config is only used for onprem deployments.

Reply:
If its not there then the call to `validate` below fails.